src/main/java/io/divolte/server/kafka/AvroRecordBufferSerializer.java

```diff
@@ -35,8 +50,26 @@ public void configure(final Map<String, ?> configs, final boolean isKey) {
         // Extract the AVRO record as a byte array.
         // (There's no way to do this without copying the array.)
         final ByteBuffer avroBuffer = data.getByteBuffer();
-        final byte[] avroBytes = new byte[avroBuffer.remaining()];
-        avroBuffer.get(avroBytes);
+        final int recordSize = avroBuffer.remaining();
+        final byte[] avroBytes;
+        final int offset;
+        switch (mode) {
+            case CONFLUENT:
+                int id = schemaId.get();
+                avroBytes = new byte[5 + recordSize];
+                avroBytes[0] = 0;
+                avroBytes[1] = (byte) ((id >> 24) & 0xff);
+                avroBytes[2] = (byte) ((id >> 16) & 0xff);
+                avroBytes[3] = (byte) ((id >> 8) & 0xff);
+                avroBytes[4] = (byte) (id & 0xff);
+                offset = 5;
```

Comment:
It should be documented that we use exactly this layout. If Confluent ever changes it, this implementation will go stale and it should be apparent from the docs that it will.